packages/jaeger-ui/src/components/DeepDependencyGraph/Header/index.test.js

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Header from './index';
+import NameSelector from './NameSelector';
+
+describe('<ListItem>', () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = shallow(<Header />);
+  });
+
+  it('renders without exploding', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('only renders the service selector if a service is not selected', () => {
```

Comment:
to me, this implies that the service selector is not rendered once a service is chosen.
a less ambiguous (IMO) test name would be `it('renders the service selector alone if a service is not selected', () => {`

Reply:
Great point.